src/sentry/preprod/api/endpoints/project_preprod_artifact_image.py

```diff
@@ -0,0 +1,148 @@
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases.project import ProjectEndpoint
+from sentry.models.project import Project
+from sentry.objectstore import preprod
+from sentry.objectstore.service import ClientError
+from sentry.preprod.analytics import PreprodArtifactApiImageEvent
+
+logger = logging.getLogger(__name__)
+
+
+def detect_image_content_type(image_data: bytes) -> str:
+    if not image_data:
+        return "application/octet-stream"
+
+    # Check magic bytes for common image formats
+    if image_data[:8] == b"\x89PNG\r\n\x1a\n":
+        return "image/png"
+    elif image_data[:3] == b"\xff\xd8\xff":
+        return "image/jpeg"
+    elif image_data[:4] == b"RIFF" and image_data[8:12] == b"WEBP":
+        return "image/webp"
+    elif image_data[:2] in (b"BM", b"BA", b"CI", b"CP", b"IC", b"PT"):
+        return "image/bmp"
+    elif image_data[:6] in (b"GIF87a", b"GIF89a"):
+        return "image/gif"
+    elif image_data[:4] == b"\x00\x00\x01\x00":
+        return "image/x-icon"
+    elif len(image_data) >= 12 and image_data[4:12] in (b"ftypavif", b"ftypavis"):
+        return "image/avif"
+    elif len(image_data) >= 12 and image_data[4:12] in (
+        b"ftypheic",
+        b"ftypheix",
+        b"ftyphevc",
+        b"ftyphevx",
+    ):
+        return "image/heic"
+
+    # Default to generic binary if we can't detect the type
+    logger.warning(
+        "Could not detect image content type from magic bytes",
+        extra={"first_bytes": image_data[:16].hex() if len(image_data) >= 16 else image_data.hex()},
+    )
+    return "application/octet-stream"
+
+
+@region_silo_endpoint
+class ProjectPreprodArtifactImageEndpoint(ProjectEndpoint):
+    owner = ApiOwner.EMERGE_TOOLS
+    publish_status = {
+        "GET": ApiPublishStatus.EXPERIMENTAL,
+    }
+
+    def get(
+        self,
+        request: Request,
+        project: Project,
+        image_id: str,
+    ) -> HttpResponse:
+
+        analytics.record(
+            PreprodArtifactApiImageEvent(
+                organization_id=project.organization_id,
+                project_id=project.id,
+                user_id=request.user.id,
+                image_id=image_id,
+            )
+        )
+
+        organization_id = project.organization_id
+        project_id = project.id
+
+        object_key = f"{organization_id}/{project_id}/{image_id}"
+        logger.info(
+            "Retrieving image from objectstore",
+            extra={
+                "organization_id": organization_id,
+                "project_id": project_id,
+                "image_id": image_id,
+            },
+        )
+        client = preprod.for_project(organization_id, project_id)
+
+        try:
+            result = client.get(object_key)
+            # Read the entire stream at once
+            image_data = result.payload.read()
```

Comment:
nit: is this necessary or is there a way to forward to stream to the HttpResponse for handling?